pkg/app/piped/executor/ecs/rollback.go

```diff
@@ -158,8 +158,14 @@ func rollback(ctx context.Context, in *executor.Input, platformProviderName stri
 			return false
 		}
 
-		if err := client.ModifyListeners(ctx, currListenerArns, routingTrafficCfg); err != nil {
-			in.LogPersister.Errorf("Failed to routing traffic to PRIMARY variant: %v", err)
+		currListenerRuleArns, err := client.GetListenerRuleArns(ctx, currListenerArns)
+		if err != nil {
+			in.LogPersister.Errorf("Failed to get current active listener rule: %v", err)
+			return false
+		}
+
+		if err := client.ModifyListenerOrRule(ctx, currListenerArns, currListenerRuleArns, routingTrafficCfg); err != nil {
```

Comment:
Instead of passing `currListenerRuleArns` to the function `ModifyListenerOrRule` (yes, the name is confusing), we should check `currListenerRuleArns > 0` in this place, and call to `ModifyRules` if the condition check passed or call to `ModifyListeners` (the previous one) if the condition return false. wdyt? 🤔

Reply:
@khanhtc1202 
Thanks for the review. I've implemented the check for currListenerRuleArns and used ModifyRules or ModifyListeners accordingly to enhance rollback precision.
[965ba41](https://github.com/pipe-cd/pipecd/pull/4668/commits/965ba41a05d1fa84ec306451fa078ef62b93c2bd)